app/controllers/reports/patient_lists_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Reports::PatientListsController < AdminController
+  skip_after_action :verify_authorized, if: -> { current_admin.permissions_v2_enabled? }
+  after_action :verify_authorization_attempted, if: -> { current_admin.permissions_v2_enabled? }
+
+  def show
+    scope = if current_admin.permissions_v2_enabled?
+      if region_class == "facility_group"
+        authorize_v2 { current_admin.accessible_facility_groups(:view_pii) }
+      else
+        authorize_v2 { current_admin.accessible_facilities(:view_pii) }
+      end
+    else
+      authorize([:cohort_report, region_class.classify.constantize], :patient_list?)
+    end
+    @region = scope.find_by!(slug: params[:id])
```

Comment:
Finally, here is the check that does the work.  If we know that the "scope" is already limited to things the admin has access too, we can rely on `find_by!` to either return something the admin has access to or raise an ActiveRecord::RecordNotFound.